components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -757,17 +757,20 @@ void AdsServiceImpl::MaybeStart(const bool should_restart) {
     return;
   }
 
+  ++total_number_of_starts_;
```

Comment:
Start/stop are only called if a user enables/disables ads in brave://rewards. Would it not be safer to disable the UI on/off switch while starting/stopping the service.

Reply:
Start can also be called when bat ads service process disconnects (for example when crashes). Here we set disconnect handler which calls MaybeStart(true):
https://github.com/brave/brave-core/blob/master/components/brave_ads/browser/ads_service_impl.cc#L723

This leads to Start() call schedule with 1 seconds delay:
https://github.com/brave/brave-core/blob/master/components/brave_ads/browser/ads_service_impl.cc#L762

When we wait for Start call, there can be another bat ads service process crash, which leads to another Start() call schedule. So we will have two Start calls without Shutdown call between them.